### PR TITLE
New Workflow - Cisco ASA - Check Host Block Status - Microsoft Teams and Slack

### PR DESCRIPTION
**Pull request by dmandyna-r7**
## Requirements

Workflow contributors, verify you have completed the following items by checking them off:

### Checklist
- [x] Workflow verified as working in a test
- [x] Workflow bundle includes `Name_of_Workflow.icon` file, `help.md` file, `workflow.spec.yaml` file, and `screenshots` directory
- [x] Workflow title should follow general [Title of Articles](http://grammar.yourdictionary.com/capitalization/rules-for-capitalization-in-titles.html) format
- [x] Workflow includes a detailed Description
- [x] Workflow step names follow [Title of Articles](http://grammar.yourdictionary.com/capitalization/rules-for-capitalization-in-titles.html) format
- [x] Triggers, Decisions, and other workflow steps include descriptions where needed
- [x] Workflow includes at least one `Final Report` artifact that summarizes workflow execution details
- [x] Workflow bundle (directory) is named after the title of the workflow in snake case with first letters capitalized, e.g. `workflows/Name_of_Workflow/`
- [x] `help.md` file includes Description, Key Features, Requirements, Setup, Technical Details, Version History, and References
- [x] `workflow.spec.yaml` file includes `products`, `name`, `title`, `description`, `use_cases` tags, `keywords` tags, `source_url`, and the `name` for each screenshot matches a PNG image in the `screenshots` directory

### Assessment

<details>

```
[*] Validating workflow at /workflows/workflows/Check_Host_Block_Status_with_Cisco_ASA_from_Slack

[*] Running Integration Validators...
[*] Executing validator WorkflowDirectoryNameMatchValidator
[*] Executing validator WorkflowFilesValidator
[*] Executing validator WorkflowHelpValidator
[*] Executing validator WorkflowChangelogValidator
[*] Executing validator WorkflowVendorValidator
[*] Executing validator WorkflowVersionValidator
[*] Executing validator WorkflowExtensionValidator
[*] Executing validator WorkflowSupportValidator
[*] Executing validator WorkflowPNGHashValidator
[*] Executing validator WorkflowICONFileNameValidator
[*] Executing validator WorkflowScreenshotValidator
[*] Executing validator WorkflowTitleValidator
[*] Executing validator WorkflowDescriptionValidator
[*] Executing validator WorkflowNameValidator
[*] Executing validator WorkflowProfanityValidator
[*] Executing validator WorkflowHelpPluginUtilizationValidator
[*] Executing validator WorkflowICONFileValidator
[*] Workflow successfully validated!

----
[*] Total time elapsed: 89.332ms

```

<summary>
icon-validate /workflows/workflows/Check_Host_Block_Status_with_Cisco_ASA_from_Slack
</summary>
</details>

<details>

```
[*] Validating workflow at /workflows/workflows/Check_Host_Block_Status_with_Cisco_ASA_from_Microsoft_Teams

[*] Running Integration Validators...
[*] Executing validator WorkflowDirectoryNameMatchValidator
[*] Executing validator WorkflowFilesValidator
[*] Executing validator WorkflowHelpValidator
[*] Executing validator WorkflowChangelogValidator
[*] Executing validator WorkflowVendorValidator
[*] Executing validator WorkflowVersionValidator
[*] Executing validator WorkflowExtensionValidator
[*] Executing validator WorkflowSupportValidator
[*] Executing validator WorkflowPNGHashValidator
[*] Executing validator WorkflowICONFileNameValidator
[*] Executing validator WorkflowScreenshotValidator
[*] Executing validator WorkflowTitleValidator
[*] Executing validator WorkflowDescriptionValidator
[*] Executing validator WorkflowNameValidator
[*] Executing validator WorkflowProfanityValidator
[*] Executing validator WorkflowHelpPluginUtilizationValidator
[*] Executing validator WorkflowICONFileValidator
[*] Workflow successfully validated!

----
[*] Total time elapsed: 67.868ms

```

<summary>
icon-validate /workflows/workflows/Check_Host_Block_Status_with_Cisco_ASA_from_Slack
</summary>
</details>

### Screenshots

Screenshots help your reviewers provide valuable feedback on your contribution. To add a screenshot here, simply drag and drop the PNG files from a file explorer.

##### Microsoft Teams
workflow1
workflow2
##### Slack
workflow1
workflow2

##### Microsoft Teams
job1
teams1
teams2

##### Slack
job1
slack1
slack2

#### Artifact
artifact1
artifact2

### CODE REVIEW

File: workflows/Check_Host_Block_Status_with_Cisco_ASA_from_Microsoft_Teams/Check_Host_Block_Status_with_Cisco_ASA_from_Microsoft_Teams.icon [hunk 1]
```diff
@@ -0,0 +1,2121 @@
...
+          },
+          "1c4cb555-b2c6-448f-bfc6-ae6097238ca4": {
+            "nodeId": "1c4cb555-b2c6-448f-bfc6-ae6097238ca4",
+            "name": "Host Blocked",
+            "type": "artifact",
+            "continueOnFailure": false,
+            "isDisabled": false,
+            "parameters": {
+              "input": {
+                "content": "🚫 🚧 The host {{[8ec756b6-1e90-41d2-b55d-f079bb84d9e5].[$item]}} is blocked. 🚧 🚫\n"
```
Comment (mrinehart-r7) [inline workflows/Check_Host_Block_Status_with_Cisco_ASA_from_Microsoft_Teams/Check_Host_Block_Status_with_Cisco_ASA_from_Microsoft_Teams.icon:429]:
```suggestion
                "content": "🚫 The host {{[8ec756b6-1e90-41d2-b55d-f079bb84d9e5].[$item]}} is blocked. 🚫\n"
```

File: workflows/Check_Host_Block_Status_with_Cisco_ASA_from_Microsoft_Teams/Check_Host_Block_Status_with_Cisco_ASA_from_Microsoft_Teams.icon [hunk 2]
```diff
@@ -0,0 +1,2121 @@
...
+          },
+          "559de269-26c7-4bc7-aa09-cb0838ea2173": {
+            "nodeId": "559de269-26c7-4bc7-aa09-cb0838ea2173",
+            "name": "Host not blocked",
+            "type": "artifact",
+            "continueOnFailure": false,
+            "isDisabled": false,
+            "parameters": {
+              "input": {
+                "content": "✅ 🚧 The host {{[8ec756b6-1e90-41d2-b55d-f079bb84d9e5].[$item]}} is not blocked. 🚧 ✅"
```
Comment (mrinehart-r7) [inline workflows/Check_Host_Block_Status_with_Cisco_ASA_from_Microsoft_Teams/Check_Host_Block_Status_with_Cisco_ASA_from_Microsoft_Teams.icon:447]:
```suggestion
                "content": "✅ The host {{[8ec756b6-1e90-41d2-b55d-f079bb84d9e5].[$item]}} is not blocked. ✅"
```

File: workflows/Check_Host_Block_Status_with_Cisco_ASA_from_Microsoft_Teams/Check_Host_Block_Status_with_Cisco_ASA_from_Microsoft_Teams.icon [hunk 3]
```diff
@@ -0,0 +1,2121 @@
...
+  "kom": {
+    "komandVersion": "1.60.1-0-g652e5",
+    "komFileVersion": "2.0.0",
+    "exportedAt": "2020-08-23T09:55:48.89417846Z",
+    "workflowVersions": [
+      {
+        "name": "Check Host Block Status with Cisco ASA from Microsoft Teams",
+        "type": "runnable",
+        "version": "",
+        "description": "",
```
Comment (jmcadams-r7) [inline workflows/Check_Host_Block_Status_with_Cisco_ASA_from_Microsoft_Teams/Check_Host_Block_Status_with_Cisco_ASA_from_Microsoft_Teams.icon:11]: Need a description here.

File: workflows/Check_Host_Block_Status_with_Cisco_ASA_from_Microsoft_Teams/help.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
+# Description
+
+This workflow accepts a Microsoft Teams command containing a list of hosts. These hosts will be checked against a pre-defined address group in Cisco ASA to determine whether or not the IP address is present. It is assumed that this address group will be used to store IP addresses for a block policy. A message with the results will be returned.
+
+Sample Microsoft Teams Trigger Commands:
+
+`!block-status 198.51.100.100`
+`!block-status example.com`
+`!block-status 198.51.100.100 example.com`
```
Comment (mrinehart-r7) [inline workflows/Check_Host_Block_Status_with_Cisco_ASA_from_Microsoft_Teams/help.md:9]:
```suggestion
`!block-status 198.51.100.100`

`!block-status example.com`

`!block-status 198.51.100.100 example.com`
```

File: workflows/Check_Host_Block_Status_with_Cisco_ASA_from_Microsoft_Teams/help.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+# Key Features
+
+* The ability to see if an IP address is currently blocked by the firewall
```
Comment (mrinehart-r7) [inline workflows/Check_Host_Block_Status_with_Cisco_ASA_from_Microsoft_Teams/help.md:13]:
```suggestion
* See if an IP address is currently blocked by the firewall
```

File: workflows/Check_Host_Block_Status_with_Cisco_ASA_from_Slack/Check_Host_Block_Status_with_Cisco_ASA_from_Slack.icon [hunk 6]
```diff
@@ -0,0 +1,1443 @@
...
+  "kom": {
+    "komandVersion": "1.60.1-0-g652e5",
+    "komFileVersion": "2.0.0",
+    "exportedAt": "2020-08-23T09:55:45.082198174Z",
+    "workflowVersions": [
+      {
+        "name": "Check Host Block Status with Cisco ASA from Slack",
+        "type": "runnable",
+        "version": "",
+        "description": "",
```
Comment (jmcadams-r7) [inline workflows/Check_Host_Block_Status_with_Cisco_ASA_from_Slack/Check_Host_Block_Status_with_Cisco_ASA_from_Slack.icon:11]: need a description here

File: workflows/Check_Host_Block_Status_with_Cisco_ASA_from_Slack/Check_Host_Block_Status_with_Cisco_ASA_from_Slack.icon [hunk 7]
```diff
@@ -0,0 +1,1443 @@
...
+          },
+          "1c4cb555-b2c6-448f-bfc6-ae6097238ca4": {
+            "nodeId": "1c4cb555-b2c6-448f-bfc6-ae6097238ca4",
+            "name": "Host Blocked",
+            "type": "artifact",
+            "continueOnFailure": false,
+            "isDisabled": false,
+            "parameters": {
+              "input": {
+                "content": "🚫 🚧 The host {{[8ec756b6-1e90-41d2-b55d-f079bb84d9e5].[$item]}} is blocked. 🚧🚫\n"
```
Comment (mrinehart-r7) [inline workflows/Check_Host_Block_Status_with_Cisco_ASA_from_Slack/Check_Host_Block_Status_with_Cisco_ASA_from_Slack.icon:245]:
```suggestion
                "content": "🚫 The host {{[8ec756b6-1e90-41d2-b55d-f079bb84d9e5].[$item]}} is blocked. 🚫\n"
```

File: workflows/Check_Host_Block_Status_with_Cisco_ASA_from_Slack/Check_Host_Block_Status_with_Cisco_ASA_from_Slack.icon [hunk 8]
```diff
@@ -0,0 +1,1443 @@
...
+          },
+          "559de269-26c7-4bc7-aa09-cb0838ea2173": {
+            "nodeId": "559de269-26c7-4bc7-aa09-cb0838ea2173",
+            "name": "Host not blocked",
+            "type": "artifact",
+            "continueOnFailure": false,
+            "isDisabled": false,
+            "parameters": {
+              "input": {
+                "content": "✅🚧 The host {{[8ec756b6-1e90-41d2-b55d-f079bb84d9e5].[$item]}} is not blocked. 🚧✅"
```
Comment (mrinehart-r7) [inline workflows/Check_Host_Block_Status_with_Cisco_ASA_from_Slack/Check_Host_Block_Status_with_Cisco_ASA_from_Slack.icon:263]:
```suggestion
                "content": "✅ The host {{[8ec756b6-1e90-41d2-b55d-f079bb84d9e5].[$item]}} is not blocked. ✅"
```

File: workflows/Check_Host_Block_Status_with_Cisco_ASA_from_Slack/help.md [hunk 9]
```diff
@@ -0,0 +1,54 @@
+# Description
+
+This workflow accepts a Slack command containing an IP address. This IP address will be checked against a pre-defined address group in Fortinet to determine if the IP address is present or not. It is assumed that this address group will be used to store IP addresses for a block policy. A message with the results will be returned.
```
Comment (mrinehart-r7) [inline workflows/Check_Host_Block_Status_with_Cisco_ASA_from_Slack/help.md:3]:
```suggestion
This workflow accepts a Slack command containing an IP address. This IP address will be checked against a pre-defined address group in Cisco ASA to determine if the IP address is present or not. It is assumed that this address group will be used to store IP addresses for a block policy. A message with the results will be returned.
```

File: workflows/Check_Host_Block_Status_with_Cisco_ASA_from_Slack/help.md [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+
+Sample Slack Trigger Commands:
+
+`@Rapid7 InsightConnect !block-status 198.51.100.100`
+`@Rapid7 InsightConnect !block-status example.com`
+`@Rapid7 InsightConnect !block-status 198.51.100.100 example.com`
```
Comment (mrinehart-r7) [inline workflows/Check_Host_Block_Status_with_Cisco_ASA_from_Slack/help.md:9]:
```suggestion
`@Rapid7 InsightConnect !block-status 198.51.100.100`

`@Rapid7 InsightConnect !block-status example.com`

`@Rapid7 InsightConnect !block-status 198.51.100.100 example.com`
```

File: workflows/Check_Host_Block_Status_with_Cisco_ASA_from_Microsoft_Teams/Check_Host_Block_Status_with_Cisco_ASA_from_Microsoft_Teams.icon [hunk 11]
```diff
@@ -0,0 +1,2121 @@
...
+              "imageData": "https://us.cdn-assets.connect.insight.rapid7.com/icons/rapid7/microsoft_teams/2.2.0/icon.png"
+            },
+            "identifier": "send_message",
+            "continueOnFailure": false,
+            "isDisabled": false,
+            "isCloud": false,
+            "parameters": {
+              "input": {
+                "channel_name": "{{[eb45a27e-1a93-4580-b5dc-f48c252db9f3].[channel_name]}}",
+                "message": "⚠️ There was an error with the workflow. Cisco ASA did not return results correctly. Check This URL for more information. ⚠️\n{{[$job].[URL]}}",
```
Comment (mrinehart-r7) [inline workflows/Check_Host_Block_Status_with_Cisco_ASA_from_Microsoft_Teams/Check_Host_Block_Status_with_Cisco_ASA_from_Microsoft_Teams.icon:217]:
```suggestion
                "message": "⚠️ There was an error with the workflow. Cisco ASA did not return results correctly. Check this URL for more information. ⚠️\n{{[$job].[URL]}}",
```

**Pull request merged**